contracts/router/SynapseRouterV2.sol

```diff
@@ -0,0 +1,361 @@
+
+        address module = _bridgeModules.get(uint256(moduleId));
+        _bridgeModules.set(uint256(moduleId), bridgeModule);
+
+        emit ModuleUpdated(moduleId, module, bridgeModule);
+    }
+
+    /// @inheritdoc IRouterV2
+    function disconnectBridgeModule(bytes32 moduleId) external onlyOwner {
+        if (!_hasModule(moduleId)) revert SynapseRouterV2__ModuleNotExists();
+
+        _bridgeModules.remove(uint256(moduleId));
+        emit ModuleDisconnected(moduleId);
+    }
+
+    /// @inheritdoc IRouterV2
+    function idToModule(bytes32 moduleId) public view returns (address bridgeModule) {
+        if (!_hasModule(moduleId)) revert SynapseRouterV2__ModuleNotExists();
+        bridgeModule = _bridgeModules.get(uint256(moduleId));
+    }
+
+    /// @inheritdoc IRouterV2
+    function moduleToId(address bridgeModule) public view returns (bytes32 moduleId) {
+        uint256 len = _bridgeModules.length();
+        for (uint256 i = 0; i < len; ++i) {
+            (uint256 key, address module) = _bridgeModules.at(i);
+            if (module == bridgeModule) {
+                moduleId = bytes32(key);
+                break;
+            }
+        }
+        if (moduleId == bytes32(0)) revert SynapseRouterV2__ModuleNotExists();
+    }
+
+    /// @inheritdoc IRouterV2
+    function getDestinationBridgeTokens(address tokenOut) external view returns (BridgeToken[] memory destTokens) {
+        destTokens = _getConnectedBridgeTokens(tokenOut, false);
+    }
+
+    /// @inheritdoc IRouterV2
+    function getOriginBridgeTokens(address tokenIn) external view returns (BridgeToken[] memory originTokens) {
+        originTokens = _getConnectedBridgeTokens(tokenIn, true);
+    }
+
+    /// @inheritdoc IRouterV2
+    function getSupportedTokens() external view returns (address[] memory supportedTokens) {
+        // supported should include all bridge tokens and any pool tokens paired with a bridge token
+        // get all possible bridge tokens from supported bridge modules
+        BridgeToken[] memory bridgeTokens = _getBridgeTokens();
+
+        // get tokens in each quoter pool
+        Pool[] memory pools = swapQuoter.allPools();
+
+        // init unflattened for aggregating supported tokens
+        uint256 count;
+        address[][] memory unflattened = new address[][](pools.length + 1);
+
+        // last index of unflattened dedicated to bridge tokens
+        unflattened[pools.length] = new address[](bridgeTokens.length);
+        for (uint256 i = 0; i < bridgeTokens.length; ++i) {
+            unflattened[pools.length][i] = bridgeTokens[i].token;
+        }
+        count += bridgeTokens.length;
+
+        // fill pool tokens up to but not including last index of unflattened
+        for (uint256 i = 0; i < pools.length; ++i) {
+            Pool memory pool = pools[i];
+            unflattened[i] = new address[](pool.tokens.length);
+
+            bool does; // whether pool.tokens does contain a bridge token
```

Comment:
Need to initiate with `unflattened[pools.length].contains(pool.lpToken);` to catch the edge case of Ethereum Stablepool having LP token as bridge token.

Reply:
hmm ok. so `pool.tokens` never contains the lp token?

Reply:
No, it always contains the list of tokens "in the pool", the same behavior as in PoolModule:

https://github.com/synapsecns/synapse-contracts/blob/601dab42bac6300aa8856f7a6f6bc740d72c8539/contracts/router/interfaces/IPoolModule.sol#L43-L49